### PR TITLE
Adds code coverage for the repo

### PR DESCRIPTION
**Pull request by owaiskazi19**
Signed-off-by: Owais Kazi <owaiskazi19@gmail.com>

### Description
Added codecov for the repo. Codecov should be enabled on app.codecov.io for this repo.

### Issues Resolved
https://github.com/opensearch-project/project-tools/issues/5

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -5,6 +5,11 @@
 require 'rubygems'
 require 'rspec'
 require 'lib/tools'
+require 'simplecov'
+SimpleCov.start
```
Comment (dblock) [inline spec/spec_helper.rb:9]: Create a file in spec/support/codecov.rb with this. It’s for better organizing and it gets loaded once.

**Review by dblock — CHANGES_REQUESTED** [review]
Thanks! Let’s add a badge to README too?

File: spec/spec_helper.rb (hunk 1 above)
Comment (owaiskazi19) [reply spec/spec_helper.rb]: Done! Trying my hands on rb for the first time.

Comment (dblock) [reply spec/spec_helper.rb]: Look at CI failures, rubocop etc

**Comment by dblock** [comment]
- Run `rubocop -a ; rubocop --auto-gen-config` and it should auto-fix RuboCop violations.
- Run `bundle install` and commit `Gemfile.lock` so that the project uses a stable version of tools.

**Comment by owaiskazi19** [comment]
Tests are not able to read the codecov token `CODECOV_TOKEN="your repo token"`. We have to enable this repo from app.codecov.io. Was this repo private before?

**Comment by dblock** [comment]
> app.codecov.io for this repo says that the repo has been deactivated. We have to enable this repo from app.codecov.io. Was this repo private before?

Ugh, @gaiksaya have you been able to resolve this for the other repo we had this problem on?
This one was never with code coverage, and as OSS it shouldn't need a token.

**Comment by gaiksaya** [comment]
> 
> Ugh, @gaiksaya have you been able to resolve this for the other repo we had this problem on? This one was private for 1 minute, and never had code coverage, and as OSS it shouldn't need a token.

Nope. Added a comment here https://community.codecov.com/t/private-public-repo-not-working/453/46?u=gaiksaya. No reply yet. Will create a new issue if I still do not see any response by EOD today.

**Comment by owaiskazi19** [comment]
I can see project-tools on codecov now
Screen Shot 2022-11-02 at 11 42 54 AM

**Comment by dblock** [comment]
And it worked! Thanks @owaiskazi19.

**Review by dblock — APPROVED** [review]

**Comment by gaiksaya** [comment]
Worked for build libraries too. Not sure what changed https://app.codecov.io/gh/opensearch-project/opensearch-build-libraries

**Pull request merged**